enclave/src/polkadex_gateway.rs

```diff
@@ -190,24 +265,56 @@ fn send_cancel_request_to_openfinex(order_uuid: &OrderUUID) -> Result<(), Gatewa
 pub fn cancel_order(
     main_account: AccountId,
     proxy_acc: Option<AccountId>,
-    order_uuid: OrderUUID,
+    cancel_order: CancelOrder,
 ) -> Result<(), GatewayError> {
     // Authenticate
     authenticate_user(main_account.clone(), proxy_acc)?;
-    // if let Ok(mutex) = load_cancel_cache_pointer() {
-    //     let mut cancel_cache: SgxMutexGuard<HashSet<OrderUUID>> = mutex.lock().unwrap();
-    //     cancel_cache.insert(order_uuid);
-    //     // Send cancel order to Openfinex API
-    // }
-    send_cancel_request_to_openfinex(&order_uuid)?;
+    let mutex = CancelOrderCache::load().map_err(|_| GatewayError::NullPointer)?;
+    let mut cache = match mutex.lock() {
+        Ok(guard) => guard,
+        Err(e) => {
+            error!("Could not acquire lock on cancel cache pointer: {}", e);
+            return Err(GatewayError::UnableToLock);
+        }
+    };
+
+    send_cancel_request_to_openfinex(cancel_order.clone(), cache.request_id() as RequestId)?;
+    cache.insert_order(cancel_order.order_id);
+
+    Ok(())
+}
+
+// /// process_cancel_order does the following
+// /// 1. Checks the orderUUID with cancel request cache
+// /// 2. Remove order from Orderbook Mirror
+// /// 3. Mutate the balances
+pub fn process_cancel_order(order_uuid: OrderUUID) -> Result<(), GatewayError> {
```

Comment:
@Gauthamastro @ksr30 could you take a look at this function and tell us which of the code is necessary? The one comment below or just the uncommented part?

Reply:
Commented part is not required unless @ksr30 has some ideas for it, @ksr30 please share your thoughts since you coded that part